types/http-aws-es/index.d.ts

```diff
@@ -18,7 +18,7 @@ declare module "elasticsearch" {
     }
 
     interface ConfigOptions {
-        amazonES?: AmazonESOptions;
+        awsConfig?: AmazonESOptions;
```

Comment:
In addition to the name change, I think the structure also changed for the options (I haven't run the JS library myself so I could be wrong about this). Here is the [documentation](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html) and the [source](https://github.com/aws/aws-sdk-js/blob/master/lib/config.d.ts). It looks like `secretKey` becamse `secretAccessKey`, `accessKey` became `accessKeyId`. For this PR you could consider removing just those two keys, or, better yet, typing the awsConfig as the AWS.Config from the d.ts in aws-sdk.